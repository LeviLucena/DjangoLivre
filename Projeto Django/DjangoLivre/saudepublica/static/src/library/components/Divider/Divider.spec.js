import { shallowMount } from '@vue/test-utils'
import BrDivider from './Divider.ce.vue'

const tamanhos = ['small', 'medium', 'large']
describe('BrDivider', () => {
  test('Deve renderizar o divider', () => {
    const wrapper = shallowMount(BrDivider)
    expect(wrapper.classes('br-divider')).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('Deve renderizar a prop vertical', () => {
    const wrapper = shallowMount(BrDivider, {
      propsData: { vertical: true },
    })
    expect(wrapper.classes()).toContain('vertical')
    expect(wrapper.element).toMatchSnapshot()
  })

  tamanhos.forEach((tamanho) => {
    test(`set size attribute ${tamanho}`, () => {
      const wrapper = shallowMount(BrDivider, {
        propsData: {
          size: tamanho,
        },
      })
      if (tamanho === 'small') {
        expect(wrapper.classes()).toContain('sm')
      } else if (tamanho === 'medium') {
        expect(wrapper.classes()).toContain('md')
      } else {
        expect(wrapper.classes()).toContain('lg')
      }
    })
  })
})
