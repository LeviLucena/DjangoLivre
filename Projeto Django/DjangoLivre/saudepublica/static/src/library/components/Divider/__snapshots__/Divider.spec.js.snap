// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrDivider Deve renderizar a prop vertical 1`] = `
<span
  class="br-divider vertical"
/>
`;

exports[`BrDivider Deve renderizar o divider 1`] = `
<span
  class="br-divider"
/>
`;
